inference-engine/tests/functional/shared_test_classes/src/read_ir/read_ir.cpp

```diff
@@ -0,0 +1,125 @@
+                                                 inputShape[0],
+                                                 inputShape[2],
+                                                 inputShape[3],
+                                                 PSROIPoolNode->get_group_size(),
+                                                 PSROIPoolNode->get_spatial_scale(),
+                                                 PSROIPoolNode->get_spatial_bins_x(),
+                                                 PSROIPoolNode->get_spatial_bins_y(),
+                                                 PSROIPoolNode->get_mode());
+        } else if (ngraph::is_type<ngraph::op::v0::ROIPooling>(node)) {
+            const auto& ROIPoolNode = std::dynamic_pointer_cast<ngraph::op::v0::ROIPooling>(node);
+            CommonTestUtils::fill_data_roi(blob->buffer(),
+                                           blob->size(),
+                                           ROIPoolNode->get_input_shape(0).front() - 1,
+                                           inputShape[2],
+                                           inputShape[3],
+                                           1.0f,
+                                           ROIPoolNode->get_method() == "max");
+        } else if (ngraph::is_type<ngraph::op::v3::ROIAlign>(node)) {
+            const auto &ROIAlignNode = std::dynamic_pointer_cast<ngraph::op::v3::ROIAlign>(node);
+            std::vector<float> blobData(ROIAlignNode->get_shape()[0] * 4);
+            ROIAlignLayerTest::fillCoordTensor(blobData,
+                                               inputShape[2],
+                                               inputShape[3],
+                                               ROIAlignNode->get_spatial_scale(),
+                                               ROIAlignNode->get_sampling_ratio(),
+                                               ROIAlignNode->get_pooled_h(),
+                                               ROIAlignNode->get_pooled_w());
+            blob = FuncTestUtils::createAndFillBlobWithFloatArray(info.getTensorDesc(), blobData.data(), blobData.size());
+        }
+    } else {
+        blob = GenerateInput(info);
+    }
+    return blob;
+}
+
+void ReadIRTest::Infer() {
+    inferRequest = executableNetwork.CreateInferRequest();
+    inputs.clear();
+
+    const auto& inputsInfo = executableNetwork.GetInputsInfo();
+    for (const auto& param : function->get_parameters()) {
+        const auto infoIt = inputsInfo.find(param->get_friendly_name());
+        GTEST_ASSERT_NE(infoIt, inputsInfo.cend());
+
+        const auto& info = infoIt->second;
+
+        InferenceEngine::Blob::Ptr blob(nullptr);
+        for (size_t i = 0; i < param->get_output_size(); i++) {
+            for (const auto& node : param->get_output_target_inputs(i)) {
+                const auto nodePtr = node.get_node()->shared_from_this();
+                if (isRoiOperation(nodePtr)) {
```

Comment:
I consider this approach not really flexible and scalable. What if we will need to have another special data generators for another operations? 
From my point of view, such functionality should be implemented in a more general way, e.g. as some kind of mapping of operation type to target generator function. 

And by the way, ROI operation are not the only ones which require special input data generators, FakeQuantize, DetectionOutpout, GatherTree and maybe some other ops require it also.

Reply:
redeveloped